code/datums/weakrefs.dm

```diff
@@ -25,6 +67,11 @@
 	target?.weak_reference = null
 	return ..()
 
+/**
+ * Retrieves the datum that this weakref is referencing.
+ *
+ * This will return `null` if the datum was deleted. This MUST be respected.
```

Comment:
Respected how? Do you mean, "the user of `.resolve()` should be aware that this can return `null`?"

Reply:
Yes.